docs/reference/ingest/processors/csv.asciidoc

```diff
@@ -0,0 +1,33 @@
+[[csv-processor]]
+=== CSV Processor
+Extracts fields from CSV line out of a single text field within a document. Any empty field in CSV will be skipped.
+
+[[csv-options]]
+.CSV Options
+[options="header"]
+|======
+| Name              | Required  | Default  | Description
+| `field`           | yes       | -        | The field to extract data from
+| `target_fields`   | yes       | -        | The array of fields to assign extracted values to
+| `separator`       | no        | ,        | Separator used in CSV, has to be single character string
+| `quote`           | no        | "        | Quote used in CSV, has to be single character string
+| `ignore_missing`  | no        | `true`   | If `true` and `field` does not exist, the processor quietly exits without modifying the document
+| `trim`            | no        | `false`  | Trim leading whitespaces in every field
```

Comment:
Would it be possible for the functionality of `trim` to be changed to trim both leading _and_ trailing whitespace?

There is a form of log file with pipe separated values and the padding usually goes at the end of the field, for example:

```
2018-01-06 16:56:14.295748|INFO    |VirtualServer |1  |listening on 0.0.0.0:9987, :::9987
2018-01-06 17:19:44.465252|DEBUG   |VirtualServer |1  |client 'User1'(id:2) changed default admin channelgroup to 'Guest'(id:8)
```

You could say that that can be handled by adding a `trim` processor for each field immediately after the `csv` processor in the ingest pipeline, and that's true, but I suspect it will cause frustration that leading whitespace can be trimmed with a single option whereas trailing whitespace for a 10 field file takes 10 additional processors.

Another reason for trimming both leading and trailing whitespace is consistency with the `find_file_structure` endpoint - search for the word "trim" in [its docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/ml-find-file-structure.html).  That endpoint treats a trim as meaning trim both leading and trailing whitespace.  After this PR is merged we will want to change `find_file_structure` to add the appropriate `csv` processor to the ingest pipeline it suggests instead of requiring that the user finds some other way to parse CSV (which they have to at the moment and is one of the reasons this PR came about).  Again, the `find_file_structure` endpoint will easily be able to add `trim` processors mechanically after the `csv` processor if appropriate, but its output will then be messier and it will draw attention to the inconsistency in the meaning of "trim".

Finally, if there is a really good reason why trim just trims leading whitespace (i.e. industry standard or something, not just that it made the coding easier), then I think the option should be called `trimStart` to emphasize that it only trims leading whitespace.